From c54f13675e3efc47ddbd090f34376d3894cacaa9 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Fri, 27 Feb 2026 12:06:48 +0100
Subject: [PATCH] ENH: Add standalone CMake preamble for VolumeResliceDriver

Add cmake_minimum_required + find_package(Slicer) preamble
so the loadable module can be built standalone against an
installed Slicer.
---
 VolumeResliceDriver/CMakeLists.txt | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/VolumeResliceDriver/CMakeLists.txt b/VolumeResliceDriver/CMakeLists.txt
index d84a1c3..73fe1a0 100644
--- a/VolumeResliceDriver/CMakeLists.txt
+++ b/VolumeResliceDriver/CMakeLists.txt
@@ -1,3 +1,21 @@
+cmake_minimum_required(VERSION 3.16.3...3.19.7 FATAL_ERROR)
+
+project(VolumeResliceDriver)
+
+set(CMAKE_CXX_STANDARD "17")
+
+find_package(Slicer REQUIRED)
+include(${Slicer_USE_FILE})
+
+if(NOT Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR)
+  set(Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR
+    "lib/Slicer-${Slicer_VERSION}/qt-loadable-modules")
+endif()
+if(NOT Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR)
+  set(Slicer_INSTALL_QTLOADABLEMODULES_SHARE_DIR
+    "share/Slicer-${Slicer_VERSION}/qt-loadable-modules")
+endif()
+
 #-----------------------------------------------------------------------------
 set(MODULE_NAME "VolumeResliceDriver") # Do not use 'project()'
 set(MODULE_TITLE "Volume Reslice Driver")
-- 
2.52.0

